sdk/protocol2_test.go

```diff
@@ -0,0 +1,169 @@
+package sdk_test
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"os"
+	"testing"
+
+	sdk_args "github.com/newrelic/infra-integrations-sdk/args"
+	"github.com/newrelic/infra-integrations-sdk/metric"
+	"github.com/newrelic/infra-integrations-sdk/sdk"
+)
+
+func TestNewIntegrationProtocol2Data(t *testing.T) {
+	i, err := sdk.NewIntegrationProtocol2("TestIntegration", "1.0", new(struct{}))
+	if err != nil {
+		t.Fatal()
+	}
+
+	if i.Name != "TestIntegration" {
+		t.Error()
+	}
+	if i.IntegrationVersion != "1.0" {
+		t.Error()
+	}
+	if i.ProtocolVersion != "2" {
+		t.Error()
+	}
+	if len(i.Data) != 0 {
+		t.Error()
+	}
+}
+
+func TestNewIntegrationProtocol2WithDefaultArguments(t *testing.T) {
+	type argumentList struct {
+		sdk_args.DefaultArgumentList
+	}
+
+	var al argumentList
+
+	os.Args = []string{"cmd"}
+	flag.CommandLine = flag.NewFlagSet("name", 0)
```

Comment:
I don't understand these 2 lines. What is their purpose? And it's not needed any check for them?

Reply:
It's copy pasted from the other protocol tests. But anyway they are not needed so I'll remove them 👍